lib/ansible/modules/cloud/azure/azure_rm_managed_disk_facts.py

```diff
@@ -125,7 +125,7 @@ def exec_module(self, **kwargs):
 
         self.results['ansible_facts']['azure_managed_disk'] = (
             self.get_item() if self.name
-            else self.list_items()
+            else (self.list_items_by_resource_group() if self.resource_group else self.list_items())
```

Comment:
Just a note, nested ternary operations are not very readable.  When you get to the point where you would nest, it's best to convert to using regular if-elif-else statements (or nested if statements) instead.